persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java

```diff
@@ -29,8 +30,9 @@ public JpaStoreConfigurationBuilder entityClass(Class<?> entityClass) {
       return self();
    }
 
+   @Deprecated
    public JpaStoreConfigurationBuilder batchSize(long batchSize) {
-      attributes.attribute(BATCH_SIZE).set(batchSize);
+      attributes.attribute(MAX_BATCH_SIZE).set((int) batchSize);
```

Comment:
I've updated the code so it handles inputs less/greater than int boundaries. It's worth noting that it's valid for the user to enter a negative max-batch-size as this indicates no batching should occur.